tests/python/contrib/test_cmsisnn/test_pooling.py

```diff
@@ -103,18 +105,17 @@ def test_op_int8(
     compiler_cpu,
     cpu_flags,
 ):
-    """Tests QNN pooling op for int8 inputs"""
+    """Tests QNN pooling op for int8 and int16 inputs"""
```

Comment:
At present, we support int8, int16 input and output for the pool. Instead of `input`, we could say `pooling`. This also suggests that we should have a negative test where dtypes of the input and output to a pool op sequence do not match. For such cases, the CMSIS-NN partitioner should fail. To be clear, (int8, int16)/(int16,int8) kind of combinations should get mapped to TVM inplace of CMSIS-NN.

Reply:
I tried to add a negative test for this but it wasn't possible to produce a model that fell into this condition, particularly in the case of max_pool. In the end I think it is okay to leave the check for this state in cmsisnn.py but I don't see an easy way to test this directly.